### ISSUE TITLE
Make it easier to use newly created types in recover blocks

### ISSUE THREAD

**Issue by yorickpeterse** [inko-lang/inko#528]
Due to unique types being a fairly new addition to Inko, pretty much all code in the standard library returns a `T` when creating new values. This can make it difficult to use such methods in `recovery` blocks. For example, `IpAddress.clone` returns an `IpAddress` even though should be fine to return an `uni IpAddress` instead. This prevents you from doing something like `recover ip.clone` unless `ip` is already a `uni IpAddress`.

In fact, I'll go as far as to say that types _should_ default to what I can best describe as "maybe uni": a type that starts out as `uni T` but automatically turns into a `T` when needed (e.g. when using a method not allowed on a `uni T`). This way the end user isn't restricted by the limitations of a `uni T` when they don't care about it, but they can still use the values in `recover` blocks if needed. This does complicate type inference, as the usage of some value requires turning it and any earlier instances of the `uni T` type into a `T` somehow.

Another option is that we simply make this the default behaviour of `uni T` values. While this removes the need for an extra kind of type, it may lead to confusing behaviour where you start a vale as `uni T`, use it in a way that requires it to be a `T` without knowing it, then end up with it not being sendable later on when you expect it to be sendable. Then again, this is also true for the previous proposal.

Long story short: `uni T` should be the default, but we should somehow support silently turning this into a `T` where needed _without_ explicit type annotations or casts.

**Comment by yorickpeterse** [comment]
Here's an example of where the current setup is annoying: `std::fs::dir.list` returns an `Array[Path]`. While we can recover that into `uni Array[Path]`, we can't recover the individual `Path` values into `uni Path`, because `Path` isn't sendable and thus can't cross the `recover` boundary. This means you can't send these paths to a process, unless you first convert them to a `String`.

**Comment by yorickpeterse** [comment]
I realised we can probably do the following:

First we change `ref uni T`/`mut uni T` to `uni ref T`/`uni mut T`. The behaviour stays the same, but this better reflects what the reference is: a unique temporary reference you can't pass around, alias, etc.

Next, in a `recover` block the outer `T`, `ref T` and `mut T` values are exposed as `ref uni T` and `mut uni T`. Fields accessed through a `uni T` receiver are exposed the same way, and so are fields typed as `uni T` (regardless of the receiver type).

For fields we then allow using them if they are typed as `ref uni T`/`mut uni T`, even though such types are not sendable. 

The rest stays the same, i.e. you can't send `ref uni T` to another process, and methods called on such references must have sendable input/output.

This should be perfectly sound due to the sendability restrictions being in place, but it makes recovering values much easier. For example, if I have a `Config` type with an `IpAddress` field, I can now do `recover config.ip.clone` and get a `uni IpAddress`. This still meets all the sendability requirements (i.e. no mutations take place that could leak references), but removes the need for complicating the type system further.
